Adding TRR for Disable or Modify M365 Logging (#30)
tired-labs/techniques: reports/trr0000/m365/README.md

# Disable or Modify Cloud Logs (Microsoft 365)

## Metadata

| Key          | Value                                |
|--------------|--------------------------------------|
| ID           | TRR0000                              |

> | External IDs | [T1562.008]                          |

Reviewer: ```suggestion
| External IDs | [T1685.002 ]                          |
```

New ID! Thanks MITRE.

> | Tactics      | Defense Evasion                      |

Reviewer 2: ```suggestion
| External IDs | [T1685.002]                          |
| Tactics      | Defense Impairment                      |
```

| Platforms    | Microsoft 365                        |
| Contributors | Andrew VanVleet                      |

### Scope Statement

This TRR covers procedures to disable or modify cloud logging from Microsoft 365
(M365) workloads. It corresponds to MITRE ATT&CK technique T1562.008 (Impair

> Defenses: Disable or Modify Cloud Logs) as it pertains to Microsoft 365.

Reviewer: ```suggestion
(M365) workloads. It corresponds to MITRE ATT&CK technique T1685.002 (Disable or Modify Tools: Disable or Modify Cloud Log ) as it pertains to Microsoft 365.
```

> Defenses: Disable or Modify Cloud Logs) as it pertains to Microsoft 365.

Reviewer 2: ```suggestion
(M365) workloads. It corresponds to MITRE ATT&CK technique [T1685.002] (Disable
or Modify Tools: Disable or Modify Cloud Log) as it pertains to Microsoft 365.
```

The following elements are out of scope for this TRR:

- Azure and Entra ID logging (diagnostic settings for Entra activity logs and
  the Azure Activity Log) are covered in a separate TRR.
- Per-workload audit controls for workloads other than Exchange. Power BI/Fabric
  and Dynamics 365/Dataverse expose their own source-level audit settings, but
  they are noted only for completeness and are not developed as procedures.
- The log destinations and their consumers (a SIEM ingesting via the Office 365

>   Management Activity API, exported archives).

Reviewer 2: ```suggestion
- The log destinations and their consumers (a SIEM ingesting via the Office 365
  Management Activity API, exported archives, etc...).
```

Mailbox auditing is the workload-level feed that emits per-mailbox actions
according to the sign-in type: owner, delegate, and administrator. Actions
logged include `MailItemsAccessed`, `HardDelete`, `SoftDelete`, `Send`, and
`FolderBind`. Mailbox auditing is on by default and the feed has three control
points in Exchange Online:

- **Organization-wide** (`Set-OrganizationConfig -AuditDisabled $true`): the
  master override for the mailbox audit feed across the tenant.
- **Per-principal bypass** (`Set-MailboxAuditBypassAssociation`): exempts
  actions performed *by* the named principal from being audited in any mailbox
  they touch.
- **Per-mailbox audited action set** (`Set-Mailbox -AuditOwner/-AuditDelegate/`
  `-AuditAdmin`): defines which actions are audited for each logon type.
  Microsoft maintains a `DefaultAuditSet`, which is collection of events that
  will be audited for each logon type by default. Customizing the logged action
  set for a logon type removes  it from the `DefaultAuditSet` and ends
  Microsoft's automatic management of it. Any newly released events must be
  added to the custom set manually. An attacker can therefore create a custom

>   set to remove high-value actions while auditing still reports as enabled.

Reviewer 2: ```suggestion
- **Per-mailbox audited action set** (`Set-Mailbox
  -AuditOwner/-AuditDelegate/-AuditAdmin`): defines which actions are audited
  for each logon type. Microsoft maintains a `DefaultAuditSet`, which is
  collection of events that will be audited for each logon type by default.
  Customizing the logged action set for a logon type removes  it from the
  `DefaultAuditSet` and ends Microsoft's automatic management of it. Any newly
  released events must be added to the custom set manually. An attacker can
  therefore create a custom set to remove high-value actions while auditing
  still reports as enabled.
```

I'm not sure if the break in the code block was intentional or not. If not, I've removed it and re-wrapped so the linter doesn't complain. If it was intentional, ignore this suggestion.

Author: I Just reverted this back in the last commit. Turns out the linter things the line started '-AuditOwner' is an improperly formated list item. So, we'll leave it with the break to keep the linter happy.

policy with a higher priority than existing policies. These can be done using
the `Set-` or `New-UnifiedAuditLogRetentionPolicy` cmdlets in Security &
Compliance Powershell.

There are two notable boundaries on this procedure:

1. Retention cannot be reduced below 90 days: `RetentionDuration` is an
enumerated value whose smallest option is `ThreeMonths`. There is no shorter
duration and no zero value. The procedure can shrink the long-term record but
cannot delete recent records outright.

2. Retention governs only the searchable UAL store (the Purview and Defender
portals, `Search-UnifiedAuditLog`, and the Audit Search Graph API). It does not
affect events streamed via the Office 365 Management Activity API. The
management API makes content blobs available for a fixed seven-day window
independent of the retention policy.

#### Detection Data Model

![DDM - Reduce Audit Log Retention](ddms/trr0000_m365_e.png)

## Available Emulation Tests

| ID             | Link |
|----------------|------|
| TRR0000.M365.A |      |
| TRR0000.M365.B |      |
| TRR0000.M365.C |      |
| TRR0000.M365.D |      |
| TRR0000.M365.E |      |

## References

> - [Disable or Modify Cloud Logs - MITRE ATT&CK]

Reviewer 2: ```suggestion
- [Disable or Modify Cloud Logs - MITRE ATT&CK][T1685.002]
```

- [Manage mailbox auditing - Microsoft Learn]
- [Set-AdminAuditLogConfig - Microsoft Learn]
- [Search the audit log - Microsoft Learn]
- [Manage audit log retention policies - Microsoft Learn]
- [Learn about auditing solutions in Microsoft Purview - Microsoft Learn]
- [Turn auditing on or off - Microsoft Learn]
- [Set-MailboxAuditBypassAssociation - Microsoft Learn]
- [Audit log activities - Microsoft Learn]

[^1]: [Office 365 Management Activity API]
[^2]: [Microsoft Purview Audit Search Graph API - Microsoft]
[^3]: [Search-MailboxAuditLog Deprecation - Microsoft]

> [T1562.008]: https://attack.mitre.org/techniques/T1562/008/

Reviewer: ```suggestion
[T1685.002]: https://attack.mitre.org/techniques/T1685/002/
```

> [Disable or Modify Cloud Logs - MITRE ATT&CK]: https://attack.mitre.org/techniques/T1562/008/

Reviewer: Redundant?

> [Disable or Modify Cloud Logs - MITRE ATT&CK]: https://attack.mitre.org/techniques/T1562/008/

Reviewer 2: ```suggestion
[T1685.002]: https://attack.mitre.org/techniques/T1685/002/
```
